lib/streamlit/report_session.py

```diff
@@ -229,6 +235,15 @@ def request_rerun(self, client_state=None):
         self._enqueue_script_request(ScriptRequest.RERUN, rerun_data)
         self._set_page_config_allowed = True
 
+    def initialize_session_state(self, initial_state: "SessionState") -> None:
```

Comment:
Instead of initializing ReportSession without a session state, then ensuring initialization in the function that retrieves it, we should be able to just initialize it from the start.

I think all the session state initialization code is remnants from when it was initialized with user-provided values.

Reply:
Ack. I didn't make this change initially since it introduces a circular import, and I wasn't sure how I felt about lazily importing `SessionState` within `ReportSession.__init__`, but thinking about it more, I think the additional simplification we get is worth it